### ISSUE TITLE
Can Sanity check be removed

### ISSUE THREAD

**Issue by Project-Magenta** [secana/PeNet#223]
I am trying to parse some PE files but I cannot access the contents of DirectoryEntries because it is null
This is because of an arbitrary limit that is never explained why we need it
Many PE files legitimately have more than 1000 resources in one directory and I cannot access those
Downgrading to 0.4.3 is not an option for me because of the very different interface and less features
[![Image of Line 124 to 133 on ImageResourceDirectory.css](https://oreotabby.com/shrz/ass/24bc6475.png)](https://github.com/secana/PeNet/blob/f682884dda619be26579416262d7e7fa3120e530/src/PeNet/Header/Pe/ImageResourceDirectory.cs#L124-L133)

I think you can also try to make a configuration option for this if you do not wish to remove it entirely

**Comment by Project-Magenta** [comment]
Temporary workaround: Patch the DLL file and place the patched version in %HOMEPATH%\.nuget\packages\penet\2.9.2\lib\net5.0

**Comment by Project-Magenta** [comment]
Alternate workaround:
```cs
using System;
using HarmonyLib;

public class PeNetSanityCheckPatcher {
    public static bool DoPatching() {
        var harmony = new Harmony("gg.none4u.sanitycheckremoval");
        harmony.PatchAll();
        return true;
    }
}

[HarmonyPatch(typeof(PeNet.Header.Pe.ImageResourceDirectory), "SanityCheckFailed")]
class Patch
{
    static bool Prefix(ref bool __result)
    {
        __result = false;
        return false;
    }
}
```

**Comment by secana** [comment]
fixed in 2.9.5
